model/custom_status.go

```diff
@@ -15,9 +16,21 @@ const (
 	MaxRecentCustomStatuses  = 5
 )
 
+var ValidCustomStatusDuration = map[string]bool{
+	"dont_clear":     true,
+	"thirty_minutes": true,
+	"one_hour":       true,
+	"four_hours":     true,
+	"today":          true,
+	"this_week":      true,
+	"date_and_time":  true,
+}
+
 type CustomStatus struct {
-	Emoji string `json:"emoji"`
-	Text  string `json:"text"`
+	Emoji     string    `json:"emoji"`
+	Text      string    `json:"text"`
+	Duration  string    `json:"duration"`
```

Comment:
Instead of modelling `Duration` and `ExpiresAt`, would we consider modelling `LastSet` and `Duration`? This gives us strictly more information, in always recording the last time a custom status was set, even if no expiration is configured. (We can then compute `ExpiresAt` trivially where needed.)

Reply:
If a user chooses a custom date/time, we would have to calculate duration based on the difference between that time and the current time. Then in the selector, we will have to calculate the expiry time based on that duration and the current time. Considering how much changes would have to be made and how much benefit would that give, I don't think we should change our approach. What do you think?

Reply:
Ah, yeah, if the experience offers directly control of the `ExpiresAt`, this approach makes sense. Thanks for the clarifications :)